apps/worker/src/app/workflow/usecases/add-job/add-delay-job.usecase.ts

```diff
@@ -36,11 +37,17 @@ export class AddDelayJob {
 
     let delay;
 
+    const subscriber = await this.subscriberRepository.findOne({
+      _id: data._subscriberId,
+      _environmentId: data._environmentId,
+    });
```

Comment:
would suggest maybe here for performance reasons, only return the "timezone" field, with the select method. Also, we can use secondarry preffered fetch